### PR TITLE
provision: "I'm already on that device??!!" flow

### PR DESCRIPTION
**Pull request by jakob223**
cc @keybase/y2ksquad 
Issue: Y2K-636

### CODE REVIEW

File: shared/provision/code-page/container.tsx [hunk 1]
```diff
@@ -26,13 +26,7 @@ export default Container.connect(
...
   (stateProps, dispatchProps, _: OwnProps) => ({
-    currentDeviceAlreadyProvisioned: stateProps.currentDeviceAlreadyProvisioned,
-    currentDeviceName: stateProps.currentDeviceName,
-    error: stateProps.error,
-    onBack: dispatchProps.onBack,
-    onSubmitTextCode: dispatchProps.onSubmitTextCode,
-    otherDeviceName: stateProps.otherDeviceName,
-    otherDeviceType: stateProps.otherDeviceType,
-    textCode: stateProps.textCode,
+    ...dispatchProps,
```
Comment (chrisnojima) [inline shared/provision/code-page/container.tsx]: Typically we've avoided doing these types of spreads as it's very easy to accidentally plumb extra props down and we currently can not detect this

File: shared/provision/code-page/index.tsx [hunk 2]
```diff
@@ -30,6 +30,7 @@ type Props = {
 type State = {
   code: string
   tab: Tab
+  troubleshooting: Boolean
```
Comment (chrisnojima) [inline shared/provision/code-page/index.tsx]: boolean

File: shared/provision/routes.tsx [hunk 3]
```diff
@@ -21,4 +21,7 @@ export const newRoutes = {
   setPublicName: {getScreen: (): typeof SetPublicName => require('./set-public-name/container').default},
   username: {getScreen: (): typeof Username => require('./username-or-email/container').default},
 }
-export const newModalRoutes = {}
+//
+// export const newModalRoutes = {
```
Comment (chrisnojima) [inline shared/provision/routes.tsx]: :fire:

File: shared/router-v2/routes.tsx [hunk 4]
```diff
@@ -5,6 +5,7 @@ import {newRoutes as gitNewRoutes, newModalRoutes as gitNewModalRoutes} from '..
 import {newRoutes as _loggedOutRoutes, newModalRoutes as loginNewModalRoutes} from '../login/routes'
 import {newRoutes as peopleNewRoutes, newModalRoutes as peopleNewModalRoutes} from '../people/routes'
 import {newRoutes as profileNewRoutes, newModalRoutes as profileNewModalRoutes} from '../profile/routes'
+import {newRoutes as provisionNewRoutes} from '../provision/routes'
```
Comment (chrisnojima) [inline shared/router-v2/routes.tsx]: Don't need this right

File: shared/provision/troubleshooting/index.tsx [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+import * as Kb from '../../common-adapters'
+import * as DevicesConstants from '../../constants/devices'
+import * as DeviceTypes from '../../constants/types/devices'
+import * as ProvisionGen from '../../actions/provision-gen'
+type Props = {
+  mode: 'QR' | 'text'
+  onCancel: () => void
+}
+
+type BigButtonProps = {onClick: () => void; icon: Kb.IconType; mainText: string; subText: string}
```
Comment (jzila) [inline shared/provision/troubleshooting/index.tsx]: these should be on their own lines.

File: shared/provision/troubleshooting/index.tsx [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+        </Kb.Box2>
+        <Kb.Box2
+          direction={Styles.isMobile ? 'vertical' : 'horizontal'}
+          style={styles.buttonBar}
+          gap="xsmall"
+        >
+          <BigButton
+            onClick={onBack}
+            icon={
+              `icon-${Styles.isMobile ? 'phone' : 'computer'}-background-${deviceIconNo}-64` as Kb.IconType
```
Comment (jzila) [inline shared/provision/troubleshooting/index.tsx]: I'd prefer if this was generated above, since it's more than just substitution.

**Pull request merged**